ntoskrnl/mm/ARM3/mminit.c

```diff
@@ -240,7 +240,7 @@ PMMPTE MiHighestUserPxe;
 #endif
 
 /* These variables define the system cache address space */
-PVOID MmSystemCacheStart = MI_SYSTEM_CACHE_START;
+PVOID MmSystemCacheStart = (PVOID)MI_SYSTEM_CACHE_START;
```

Comment:
[Line 23 of ntoskrnl/include/internal/i386/mm.h](https://git.reactos.org/?p=reactos.git;a=blob;f=ntoskrnl/include/internal/i386/mm.h;hb=a6d2172cccd5262d1a7e8a82693ab73469274b14#l23) you have:
`#define MI_SYSTEM_CACHE_START                   (PVOID)0xC1000000`
but [line 25 of ntoskrnl/include/internal/amd64/mm.h](https://git.reactos.org/?p=reactos.git;a=blob;f=ntoskrnl/include/internal/amd64/mm.h;hb=a6d2172cccd5262d1a7e8a82693ab73469274b14#l25) you have:
`#define MI_SYSTEM_CACHE_START                  0xFFFFF98000000000ULL`
so I was wondering, whether putting the cast inside that header wouldn't be better?

Reply:
I generally hate having the casts with the constants. Sadly it's a huge mess all over.

Reply:
@tkreuzer and others: this question may be in C-newbie style, but please, what is the root cause for the code portable between x86 and x86-64 to get messed up like that?

Is it:
* that some rewrite / early planning had helped?
* compilers dependent?
* SDK/DDK headers induced?